### ISSUE TITLE
Tips for doing a A-MoreOOP

### ISSUE THREAD

**Issue by DarrenCsAcc** [nus-cs2103-AY2223S1/forum#180]
Hi all,

I am currently quite behind for my IP so i am trying my best to catch up for now with the remaining time i have left. i am embarrassed to say i am currently at the stage of doing A-MoreOOP of week 3 and i am not sure how to properly organise my code for the parser part and UI part to incorporate to my main.I was wondering if anyone can give me some suggestions(example will be helpful maybe small snippets)

Thank you all for your time.

**Comment by sltsheryl** [comment]
Hi I talked abit about this here https://github.com/nus-cs2103-AY2223S1/forum/issues/91.
I think we can think of it as handling different concerns in different classes. 
So the parser, as the name suggests, could handle taking in the user input and deciding which command to call and its argument. 
While, the UI could handle displaying user interface messages such as the different task messages, welcome and bye messages.

**Comment by RezwanArefin01** [comment]
I would suggest you to go to the [iP Code Dashboard](https://nus-cs2103-ay2223s1.github.io/ip-dashboard/) and look at how others have organized their repository. It might be useful to sort the dashboard by contributions. _Generally_ the ones with more lines of code have better organization.

**Comment by nehcuy** [comment]
A little additional advice (since you only have a few days left) is to run through every minimum requirement instead of spending too much time overcustomising your bot. (Totally not what I did 🥲)

**Comment by hoang227** [comment]
In your main method, we know that the reading of the user input is a loop, just think about every step that happens in a single iteration of the loop and go from there. 

Think of the process like some sort of baton passing (so one person is holding the baton at one time). 

For example, when the user input a command, the baton will be passed to the parser with will try to comprehend what the user is trying to say and communicate that to the next person. 

Once the parser is done it will pass the baton to someone else. If the parser detected an add command , there should be some sort of task list manager taking the baton and help you add this new task into the list. 

Then once the adding is done the baton is passed to the UI which will print the out put on the screen to communicate with the user the status of his request. Then you can pass the baton back to parser to read the next input.



Hope this helps!

**Comment by rylzxc** [comment]
I referred to [this](https://github.com/se-edu/addressbook-level2/tree/master/src/seedu/addressbook) while organising my code. It points to the code base of addressbook-level2 and it is quite similar to what we're doing for iP.

**Comment by damithc** [comment]
> Think of the process like some sort of baton passing (so one person is holding the baton at one time).
> 
> For example, when the user input a command, the baton will be passed to the parser with will try to comprehend what the user is trying to say and communicate that to the next person.

Be careful not to push the baton passing metaphor too far though. One can argue that the Parser's job is to parse the command only i.e., Parser should not know who should take the baton next.

**Comment by DarrenCsAcc** [comment]
Thank you all i will work on it!

**Comment by DarrenCsAcc** [comment]
Hi all I manage to get it done. Although it's not as good as everyone's but thank you all for your help!
